src/appPackageManifest.ts

```diff
@@ -88,15 +88,30 @@ class AppPackageManifestTransform extends Transform {
     if (this.projectConfig.appType === AppType.APP) {
       const appConfig = this.projectConfig;
 
-      if (appConfig.tiles !== undefined) {
+      if (appConfig.tiles !== undefined && this.hasNative) {
+        const watchComponents =
+          this.components.watch !== undefined
+            ? Object.keys(this.components.watch)
+            : ([] as string[]);
+
         this.components.tiles = appConfig.tiles.map((tile) => ({
           name: tile.name,
           id: tile.uuid,
           platforms:
             tile.buildTargets !== undefined
-              ? tile.buildTargets
-              : appConfig.buildTargets,
+              ? tile.buildTargets.filter((target) =>
+                  watchComponents.includes(target),
+                )
+              : watchComponents,
         }));
+
+        // Remove tiles with 0 platforms
+        this.components.tiles = this.components.tiles.filter(
+          (tile) => tile.platforms.length > 0,
+        );
+        if (this.components.tiles.length === 0) {
+          this.components.tiles = undefined;
```

Comment:
Can you verify this doesn't include `tiles: undefined` key in the JSON? Having a key that is defined but has value `undefined` is different than a key that is actually not defined.

Reply:
Checked this code path and the resulting `manifest.json` doesn't include the key `tiles` in it.
Do you want me to make any other checks?

Reply:
Nope that covers it, thanks! I vaguely recalled some case similar from a while ago causing an issue like that but it must've been slightly different.